setuputil.py

```diff
@@ -18,6 +18,12 @@
 
 from pkg_resources import safe_name, parse_requirements
 
+_extra_client_requirements = []
+_extra_server_requirements = []
+
+if sys.version_info[:2] == (3, 6):
+    _extra_client_requirements.append('dataclasses')
+    _extra_server_requirements.append('dataclasses')
```

Comment:
TODO: rename client->common and don't append to server list.

Reply:
Info: We use markers in the requirements file to include packages in specific environments: https://github.com/rucio/rucio/commit/c301af509ba7eed5e34b644a68db39fcc2fa7553 : e.g. the mock package that got removed.

This work because we parse the requirements file with the `pkg_resources` package. [https://setuptools.pypa.io/en/latest/pkg_resources.html](https://setuptools.pypa.io/en/latest/pkg_resources.html#requirements-parsing)

As far as I can tell, you don't need to introduce the different lists. All elements in the install_requires of the client are also in the server.
```python
server_requirements_table = {
    'install_requires': clients_requirements_table['install_requires'] + [
```